### PR TITLE
Fix/Remove broken links

### PR DESCRIPTION
**Pull request by steffenschroeder**
https://github.com/pytest-dev/pytest/issues/1722 reported broken links in CHANGELOG.
This PR fixes or removes many broken links.
This is prerequisite for a follow-up PR which adds [linkcheck](https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-linkcheck-builder)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3575,7 +3575,7 @@ Bug Fixes
   Thanks `@sirex`_ for the report and `@nicoddemus`_ for the PR.
 
 * Replace ``raise StopIteration`` usages in the code by simple ``returns`` to finish generators, in accordance to `PEP-479`_ (`#2160`_).
-  Thanks `@tgoodlet`_ for the report and `@nicoddemus`_ for the PR.
```
Comment (bluetech) [inline CHANGELOG.rst]: Perhaps leave the name, just not linkified?

Everything else looks good to me!

File: CHANGELOG.rst [hunk 2]
```diff
@@ -3811,7 +3811,6 @@ Bug Fixes
 .. _@philpep: https://github.com/philpep
 .. _@raquel-ucl: https://github.com/raquel-ucl
 .. _@axil: https://github.com/axil
-.. _@tgoodlet: https://github.com/tgoodlet
```
Comment (asottile) [inline CHANGELOG.rst]: fwiw, now @goodboy

Comment (goodboy) [reply CHANGELOG.rst]: Heh, yeah sorry about any confusion on that one.
I finally got the username I always wanted a few months back.

File: CHANGELOG.rst (hunk 1 above)
Comment (goodboy) [reply CHANGELOG.rst]: Yeah feel free to link to my new account if you'd like.

**Pull request merged**